storage/innobase/xtrabackup/test/t/bug1394493.sh

```diff
@@ -0,0 +1,103 @@
+############################################################################
+# Bug1394493: need warning when encountering TokuDB tables
+############################################################################
+
+. inc/common.sh
+
+require_server_version_higher_than 5.7.0
+require_xtradb
+
+# Modified piece from ps_tokudb_admin
+SCRIPT_PWD=$(dirname ${MYSQLD})
+JEMALLOC_LOCATION=""
+HOTBACKUP_LOCATION=""
+
+# Check location for libjemalloc.so.1
+printf "Checking location of jemalloc library ...\n"
+for libjemall in "${SCRIPT_PWD}/lib/mysql" "/usr/lib64" "/usr/lib/x86_64-linux-gnu" "/usr/lib"; do
+  if [ -r "$libjemall/libjemalloc.so.1" ]; then
+    JEMALLOC_LOCATION="$libjemall/libjemalloc.so.1"
+    break
+  fi
+done
+if [ -z $JEMALLOC_LOCATION ]; then
+  printf "ERROR: Cannot find libjemalloc.so.1 library. Make sure you have libjemalloc1 on debian|ubuntu or jemalloc on centos package installed.\n\n";
+  skip_test "needs TokuDB"
+else
+  printf "INFO: Using jemalloc library from $JEMALLOC_LOCATION\n\n";
+fi
+
+# Check location for libHotBackup.so
+printf "Checking location of TokuBackup library ...\n"
+for libhotbackup in "$(dirname ${SCRIPT_PWD})/lib" "/usr/lib64" "/usr/lib/x86_64-linux-gnu" "/usr/lib" "${SCRIPT_PWD}/lib/mysql" "/usr/lib64/mysql" "/usr/lib/x86_64-linux-gnu/mysql" "/usr/lib/mysql"; do
+  if [ -r "$libhotbackup/libHotBackup.so" ]; then
+    HOTBACKUP_LOCATION="$libhotbackup/libHotBackup.so"
+    break
+  fi
+done
+if [ -z $HOTBACKUP_LOCATION ]; then
+  printf "ERROR: Cannot find libHotBackup.so library. Make sure you have tokudb package installed.\n\n";
+  skip_test "needs TokuDB"
+else
+  printf "INFO: Using TokuBackup library from $HOTBACKUP_LOCATION\n\n";
+fi
+
+OLD_LD_PRELOAD=${LD_PRELOAD:-""}
+export LD_PRELOAD="$JEMALLOC_LOCATION:$HOTBACKUP_LOCATION:$OLD_LD_PRELOAD"
+vlog "Set LD_PRELOAD=$LD_PRELOAD"
+export TOKU_HUGE_PAGES_OK=1
+start_server
+export LD_PRELOAD="$OLD_LD_PRELOAD"
+
+mysql <<EOF
```

Comment:
Sure, that code has to be moved to the `common.sh`. However, I find it more convenient and obvious to have a function `start_server_with_tokudb()`. Function allows not to pollute script's  environment with `LD_PRELOAD` and other TokuDB-specific stuff.